h2o-core/src/main/java/hex/Model.java

```diff
@@ -399,6 +399,26 @@ public long getOrMakeRealSeed(){
     /** @return the validation frame instance, or null
      *  if a validation frame was not specified */
     public final Frame valid() { return _valid==null ? null : _valid.get(); }
+
+    /**
+     * @return all Frames used by these params
+     */
+    public final Map<String, Frame> getAllFrames() {
+      Map<String, Frame> framesMap = new HashMap<>();
+      collectAllFrames(framesMap);
+      return framesMap;
+    }
+    
+    protected void collectAllFrames(Map<String, Frame> map) {
```

Comment:
Knit/opinion based:

- The collection of two basic frames (validation and train) could be part of the method's contract in JavaDoc, **or** there could be yet another method `collectCommonFrames`, which would make it obvious for overriders what is done. and could selectively pick functionaly to override. Calling `super` is fine to me as well.

- MIssingJavaDoc.